tests/org.eclipse.n4js.n4jsx.lang.tests/src/org/eclipse/n4js/n4jsx/lang/tests/parser/JSXWithOutFreeTextTest.xtend

```diff
@@ -117,18 +167,67 @@ class JSXWithOutFreeTextTest extends AbstractN4JSXParserTest {
 			<div></div>
 		'''.parseWithError
 	}
+	
+	/**
+	 * The example can be parsed successfully with babel, too.
+	 */
+	@Test
+	def void testRegExAmbiguityContra_01() {
+		'''
+			"Hello"
+
+			(<div></div>)
+		'''.parseSuccessfully
+	}
 
 	/**
-	 * The example produces the same error when using the babel transpiler.
+	 * The example can be parsed successfully with babel, too.
 	 */
 	@Test
-	def void testRegExAmbiguityContra() {
+	def void testRegExAmbiguityContra_02() {
 		'''
 			<div></div>
 
 			"Hello"
 		'''.parseSuccessfully
 	}
-
+	
+	@Test
+	def void testInvalidAttributeNames_01() {
+		'''
+			<a attr/**/-attr></a>
+		'''.parseWithError("JSX attribute names may not contain whitespace or comments.")
+	}
+	
+	@Test
+	def void testInvalidAttributeNames_02() {
+		'''
+			<a attr -attr></a>
+		'''.parseWithError("JSX attribute names may not contain whitespace or comments.")
+	}
+	
+	@Test
+	def void testInvalidAttributeNames_03() {
+		'''
+			<a attr-- attr></a>
```

Comment:
Do you have a successful test case like this: `<a attr--=true attr></a>`?

Reply:
I'll add a testcase

Reply:
Nice one: `-='  is a keyword so `attr-=""` cannot be parsed successfully. :+1: